www/scripts/productlist/ListOfProducts.js

```diff
@@ -213,20 +214,33 @@ function (declare, domClass, domConstruct, ItemFileWriteStore, topic,
         + name[0].toUpperCase()
         + '</span></div>';
 
-      if (!item.connected || !item.accessible) {
+      var dbStatus = item.databaseStatus;
+
+      dbStatusMsg = util.dbStatusFromCodeToString(dbStatus);
+
+      if (dbStatus !== DBStatus.OK || !item.accessible) {
         name = '<span class="product-error">'
           + name + '</span>';
 
-        if (!item.connected) {
-          statusIcon = '<span class="customIcon product-error"></span>';
-          description = '<span class="product-description-error database">'
-            + 'The database connection for this product could not be made!'
-            + '</span><br />' + description;
-        } else if (!item.accessible) {
+        if (!item.accessible) {
           statusIcon = '<span class="customIcon product-noaccess"></span>';
           description = '<span class="product-description-error access">'
             + 'You do not have access to this product!'
             + '</span><br />' + description;
+        } else if (dbStatus !== DBStatus.OK) {
+
+          var upgradeMsg = '';
+          statusIcon = '<span class="customIcon product-error"></span>';
+
+          if(dbStatus === DBStatus.SCHEMA_MISMATCH_OK ||
+             dbStatus === DBStatus.SCHEMA_MISSING){
+            upgradeMsg = ' (use server command for schema '
```

Comment:
I think we should put server between a `<tt>` to indicate it's a command.

Reply:
HTML <tt> [is not supported](https://www.w3schools.com/tags/tag_tt.asp) in HTML5.

Reply:
One of the things which are *officially* not supported but *unofficially* forever will be. Then let's use `<samp>` or `<kbd>` instead, which give teletype format.